src/ios/AppDelegate+notification.m

```diff
@@ -182,22 +253,254 @@ - (void)pushPluginOnApplicationDidBecomeActive:(NSNotification *)notification {
         self.coldstart = [NSNumber numberWithBool:NO];
         [pushHandler performSelectorOnMainThread:@selector(notificationReceived) withObject:pushHandler waitUntilDone:NO];
     }
-
-    [[NSNotificationCenter defaultCenter] postNotificationName:pushPluginApplicationDidBecomeActiveNotification object:nil];
+        
 }
 
 - (void)userNotificationCenter:(UNUserNotificationCenter *)center
        willPresentNotification:(UNNotification *)notification
          withCompletionHandler:(void (^)(UNNotificationPresentationOptions options))completionHandler
 {
     NSLog( @"NotificationCenter Handle push from foreground" );
-    // custom code to handle push while app is in the foreground
+    
     PushPlugin *pushHandler = [self getCommandInstance:@"PushNotification"];
     pushHandler.notificationMessage = notification.request.content.userInfo;
     pushHandler.isInline = YES;
+    // Add a custom flag that will be validaded in the CORES Mobile app to indicate this notification was received without any tap event.
+    pushHandler.isTapped = NO;
     [pushHandler notificationReceived];
 
-    completionHandler(UNNotificationPresentationOptionNone);
+    // Cordova Plugin Push issue #3: Always displays the notification in the Notification Center of the phone even if the app is in foreground.
+    // https://github.com/TransformativeMed/cordova-plugin-push/issues/3
+    // Display the notification on the OS tray including the sound and badge number contained in the push notification.
+    completionHandler(UNNotificationPresentationOptionAlert | UNNotificationPresentationOptionSound | UNNotificationPresentationOptionBadge);
+}
+
+// Cordova Push Plugin issues #3. Get the data of the pressed notification (single or grouped notifications) to send it to the CORES Mobile app.
+// https://github.com/TransformativeMed/cordova-plugin-push/issues/3
+- (void) sendResponseDataOfPressedNotification: (NSDictionary *)userInfo : (PushPlugin *) pushHandler {
+    
+    // Check if the notification was sent by the APNS (triggered by the CORES API) or it is a local push notification created by us to let the user open all notifications ("open all" notification).
+    if ( [userInfo objectForKey:@"coresPayload"] ) {
```

Comment:
I'm a little concerned by the way this is being designed. By putting logic in here that is specific to CORES Mobile, we are essentially being committed to maintaining and enhancing Java code indefinitely into the future. Additionally, code like this will prevent us from being able to submit these changes to the upstream repo and will make it more difficult for us to pull down changes from the upstream repo. Overall, it's going to increase the difficulty of maintenance in the long-term.

Is there a way that you can design this functionality generically so that it could be used by any mobile application? So that more of the business logic is kept inside of CORES Mobile instead of this universal plugin.